Sources/Brave/Frontend/Browser/TabManager.swift

```diff
@@ -1208,16 +1208,21 @@ extension TabManager: NSFetchedResultsControllerDelegate {
       tabsForDomain.forEach { tab in
         Task { @MainActor in
           let privateMode = PrivateBrowsingManager.shared.isPrivateBrowsing
+          guard let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: privateMode) else {
+            return
+          }
+          let allAccounts = await keyringService.allAccounts()
           // iOS does not have `HostContentSettingsMap`, so we must
           // implement `SolanaProviderImpl::OnContentSettingChanged`
-          if let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: privateMode),
-             let selectedSolAccount = await keyringService.selectedAccount(.sol),
-             tab.isSolanaAccountConnected(selectedSolAccount), // currently connected
-             !tab.isAccountAllowed(.sol, account: selectedSolAccount) { // user revoked access
+          if let selectedSolAccount = allAccounts.solDappSelectedAccount,
+             tab.isSolanaAccountConnected(selectedSolAccount.address), // currently connected
+             !tab.isAccountAllowed(.sol, account: selectedSolAccount.address) { // user revoked access
             tab.walletSolProvider?.disconnect()
           }
-          let accounts = await tab.allowedAccountsForCurrentCoin()
-          tab.accountsChangedEvent(Array(accounts))
```

Comment:
are these two lines replaced by the code below. so its eth account specifically?

Reply:
`accountsChangedEvent` is specific to Ethereum: https://github.com/search?q=repo%3Abrave%2Fbrave-core%20accountsChangedEvent&type=code

I think this was missed updating to exclude Solana accounts when adding Solana DApp support.